yt_dlp/extractor/subsplash.py

```diff
@@ -0,0 +1,184 @@
+import functools
+import math
+
+from .common import InfoExtractor
+from ..utils import (
+    InAdvancePagedList,
+    int_or_none,
+    parse_iso8601,
+    try_call,
+    url_or_none,
+)
+from ..utils.traversal import traverse_obj
+
+
+class SubsplashBaseIE(InfoExtractor):
+    def _get_headers(self, url, display_id):
+        token = try_call(lambda: self._get_cookies(url)['ss-token-guest'].value)
+        if not token:
+            webpage, urlh = self._download_webpage_handle(url, display_id)
+            token = (
+                try_call(lambda: self._get_cookies(url)['ss-token-guest'].value)
+                or urlh.get_header('x-api-token')
+                or self._search_json(
+                    r'<script[^>]+\bid="shoebox-tokens"[^>]*>', webpage, 'shoebox tokens',
+                    display_id, default={}).get('apiToken')
+                or self._search_regex(r'\\"tokens\\":{\\"guest\\":\\"([A-Za-z0-9._-]+)\\"', webpage, 'token', default=None))
+
+        if not token:
+            return None
```

Comment:
oh there was another thing I forgot to ask about:

can requests be successfully made without the token? should the last `_search_regex` call just be fatal?

Reply:
All the examples I tested required a token, but I wasn't sure if a token is required for everything, that's why I left it non-fatal.

Reply:
can we warn here then?
```suggestion
                or self._search_regex(r'\\"tokens\\":{\\"guest\\":\\"([A-Za-z0-9._-]+)\\"', webpage, 'token', default=None))

        if not token:
            self.report_warning('Unable to extract auth token')
            return None
```

Reply:
or could do it like this
```suggestion
                or self._search_regex(r'\\"tokens\\":{\\"guest\\":\\"([A-Za-z0-9._-]+)\\"', webpage, 'token', fatal=False

        if not token:
            return None
```

Reply:
Went with the first option as it more clearly shows that we are raising a warning. The fatal could easily be overlooked if other token extractions are ever added.